packages/coinstac-ui/app/render/components/runs/effects/useStartDecentralizedRun.jsx

```diff
@@ -0,0 +1,38 @@
+import { useRef, useEffect } from 'react';
+import { useQuery, useSubscription } from '@apollo/client';
+import { get } from 'lodash';
+import { useSelector, useDispatch } from 'react-redux';
+
+import { FETCH_ALL_CONSORTIA_QUERY, RUN_STARTED_SUBSCRIPTION } from '../../../state/graphql/functions';
+import { startRun } from '../../../state/ducks/runs';
+
+/*
+ * This effect checks if there are runs that must be started locally when the app first starts.
+ */
+function useStartDecentralizedRun() {
+  const dispatch = useDispatch();
+
+  const auth = useSelector(state => state.auth);
+
+  const { data } = useQuery(FETCH_ALL_CONSORTIA_QUERY);
+  const { data: runSubData } = useSubscription(RUN_STARTED_SUBSCRIPTION, {
+    variables: {
+      userId: auth.user.id,
+    },
+    skip: !auth || !auth.user || !auth.user.id,
+  });
+
+  const consortia = get(data, 'fetchAllConsortia');
+
+  useEffect(() => {
+    const run = get(runSubData, 'runStarted');
+
+    if (!run || run.status !== 'started') return;
+
+    const consortium = consortia.find(c => c.id === run.consortiumId);
+
+    dispatch(startRun(run, consortium));
+  }, [runSubData]);
+}
```

Comment:
Not sure this is the PR for it, but one difficulty is when the run errors before the pipeline actually runs causing a local error but not propagating the error out, this run then gets continually restarted->crashes

Reply:
this should not happen anymore because this one will try to run only once when the `runStarted` subscription pushes data to the client

Reply:
Cool